bitnami/grafana-alloy/templates/service.yaml

```diff
@@ -0,0 +1,55 @@
+{{- /*
+Copyright Broadcom, Inc. All Rights Reserved.
+SPDX-License-Identifier: APACHE-2.0
+*/}}
+
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ template "common.names.fullname" . }}
+  namespace: {{ include "common.names.namespace" . | quote }}
+  labels: {{- include "common.labels.standard" ( dict "customLabels" .Values.commonLabels "context" $ ) | nindent 4 }}
+    app.kubernetes.io/component: alloy
+    app.kubernetes.io/part-of: grafana-alloy
+    prometheus.io/scrape: "true"
```

Comment:
This kind of labels/annotations we set it only if metrics.enabled=true. You can try searching for defaultMetricsAnnotations in other charts